openpgp/packet/opaque.go

```diff
@@ -152,11 +157,13 @@ Truncated:
 
 func (osp *OpaqueSubpacket) Serialize(w io.Writer) (err error) {
 	buf := make([]byte, 6)
-	n := serializeSubpacketLength(buf, len(osp.Contents)+1)
+	copy(buf, osp.LengthAndContents[0])
+	n := len(osp.LengthAndContents[0])
+
 	buf[n] = osp.SubType
 	if _, err = w.Write(buf[:n+1]); err != nil {
 		return
 	}
-	_, err = w.Write(osp.Contents)
+	_, err = w.Write(osp.LengthAndContents[1])
```

Comment:
Replace this entire function with:
```suggestion
	_, err = w.Write(osp.LengthAndContents)
```